sdk/nodejs/appservice/zMixins_http.ts

```diff
@@ -106,9 +140,12 @@ export class HttpEventSubscription extends mod.EventSubscription<mod.Context<Htt
             resourceGroupName,
         }, opts);
 
-        this.url = args.route
-            ? pulumi.interpolate`https://${this.functionApp.defaultHostname}/api/{${args.route}}`
-            : pulumi.interpolate`https://${this.functionApp.defaultHostname}/api/${name}`;
+        const routePrefix = args.hostSettings && args.hostSettings.extensions && args.hostSettings.extensions.http.routePrefix;
+        const rootPath = routePrefix === "" ? "" : `${routePrefix === undefined ? "api" : routePrefix}/`;
+
+        const functionPath = args.route === undefined ? name : `{${args.route}}`;
+
+        this.url = pulumi.interpolate`https://${this.functionApp.defaultHostname}/${rootPath}${functionPath}`;

```

Comment:
dumb question.  instead of us computing this ourselves, is it possible to obtain off of the FunctionApp in some way?  If not, this is fine.  but it would be great if we could just defer to Azure here instead of computing it ourselves.

Reply:
I don't think we can get it directly. We'd need to do [something like this](https://stackoverflow.com/a/42940281/1171619).

Reply:
👍